### PR TITLE
Reduced usage of file reading

### LINKED ISSUE #155: GScan should only read files from disks once
Currently there are multiple places, where we read files from disk. This can and should be improved:

1. We're reading the content of all `*.hbs` files [here](https://github.com/TryGhost/gscan/blob/master/lib/read-theme.js#L73)
2. We're reading the content of `package.json` [here](https://github.com/TryGhost/gscan/blob/master/lib/checks/010-package-json.js#L138)
3. We're reading the content of all  `*.css` files [here](https://github.com/TryGhost/gscan/blob/master/lib/checks/001-deprecations.js#L43) and [here](https://github.com/TryGhost/gscan/blob/master/lib/checks/001-deprecations.js#L43)

### PR DESCRIPTION
**Pull request by AileenCGN**
closes #155

- Make content of CSS files and package.json file available in our theme object to reduce further disk access

### CODE REVIEW

File: lib/read-theme.js [hunk 1]
```diff
@@ -66,10 +66,17 @@ readThemeStructure = function readThemeFiles(themePath, subPath, arr) {
 };
 
 readHbsFiles = function readHbsFiles(theme) {
```
Comment (kirrg001) [inline lib/read-theme.js]: Maybe rename the function?

File: lib/read-theme.js [hunk 2]
```diff
@@ -66,10 +66,17 @@ readThemeStructure = function readThemeFiles(themePath, subPath, arr) {
...
+            return themeFile.ext.match(/\.hbs|\.css/) || themeFile.file.match(/package.json/);
+        }
+    });
     // Setup a partials array
     theme.partials = [];
 
-    return Promise.map(_.filter(theme.files, {ext: '.hbs'}), function (themeFile) {
+    // CASE: we need the actual content of all css, hbs files, and package.json for
+    // our checks
+    return Promise.map(themeFilesContent, function (themeFile) {
```
Comment (kirrg001) [inline lib/read-theme.js]: > themeFile.compiled = readHbs.handlebars.precompile(content);

We don't even have to try to precompile if it's not a css file.
Maybe use two functions? `readHbsFiles` and `readCssFiles`?

**Comment by AileenCGN** [comment]
@kirrg001 Can you have a look at this again?

**Comment by kirrg001** [comment]
@AileenCGN Will try today.

**Comment by AileenCGN** [comment]
@kirrg001 ping!

**Comment by kirrg001** [comment]
@gargol Sorry! Will look at all GScan pull requests as part of the next cycle.

**Comment by gargol** [comment]
@kirrg001 think you meant to mention @AileenCGN :wink:  
Fyi @AileenCGN as @kirrg001 mentioned, all PR's are being tracked to be reviewed and merged/closed as part of the cycle plan

**Comment by AileenCGN** [comment]
Just cleaning up here too 😉

**Comment by gargol** [comment]
Landed in master as 4ce43d480e393acf28d25ef97cb69e0c6b0ef7e2 :ship:

**Pull request closed without merging**